processor/attributesprocessor/attributes_metric_test.go

```diff
@@ -429,6 +429,63 @@ func TestMetricAttributes_Hash(t *testing.T) {
 		runIndividualMetricTestCase(t, tc, mp)
 	}
 }
+func TestMetricAttributes_Convert(t *testing.T) {
+	testCases := []metricTestCase{
+		{
+			name: "String to int (good)",
+			inputAttributes: map[string]pdata.AttributeValue{
+				"to.int": pdata.NewAttributeValueString("123"),
+			},
+			expectedAttributes: map[string]pdata.AttributeValue{
+				"to.int": pdata.NewAttributeValueInt(123),
+			},
+		},
+		{
+			name: "String to int (bad)",
+			inputAttributes: map[string]pdata.AttributeValue{
+				"to.int": pdata.NewAttributeValueString("int-10"),
+			},
+			expectedAttributes: map[string]pdata.AttributeValue{
+				"to.int": pdata.NewAttributeValueString("int-10"),
+			},
+		},
+		{
+			name: "String to double",
+			inputAttributes: map[string]pdata.AttributeValue{
+				"to.int": pdata.NewAttributeValueString("3.14.e2"),
```

Comment:
accidentally typod here and it seems like if I run `go test` inside `processor/attributesprocessor` only the trace test assertions are actually verified?? Tried breaking other tests in this file, and it passes tests fine... will look properly in a bit

Reply:
I've 'fixed' these tests now (they actually test something 😅). Unfortunately, there were 6 failing tests... I've commented these out for now...

Reply:
OK so those are failing for a few reasons...:
- Some tests are either misnamed or miswritten (i.e. `..._FilterMetricsByName` don't actually use name matchers at all)
- `Resource` matching for metrics actually has a bug...

Reply:
I've tweaked the tests that looked like they should've been tests for the metric names. They now pass. I haven't attempted to fix the final tests (`TestAttributes_FilterMetrics`) -- that looks like an actual bug (the filter is instantiated by `Resource` rather than the `Attribute` field, but tests against the attributes on the individual metrics, rather than on the attributes on the resource. I guess what's really desired is ultimately to support bother filters, but these test changes are already getting rather out of scope for the pr so I'm not planning on addressing it)

Reply:
I think for the `Resource` vs `Attribute` config being instantiated the test is actually [using a testcase name](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/attributesprocessor/attributes_metric_test.go#L55) as the value (which might be confusing)

Reply:
There's nowhere in the attributesprocessor code where attributes on metrics resources are ever checked for inclusion/exclusion which is _part_ of the reason why this breaks...

Reply:
Ahh, I got it now. Thank you for explanation. I think we need to fix the original issue then, perhaps in a separate PR to keep it separated

Reply:
Yeah I have a separate branch where I'm trying to fix that, but I don't feel like it should be a blocker for this pr? It's a bit more involved

Reply:
Opened a separate issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8393 

I think it might be not blocker, we would need to cleanup the tests though (e.g. the commented sections). If that might help, I could work on the issue with `genericMetricData` as well, just let me know

Reply:
If it would help to get this in, I can remove the commented sections (or the entire `TestAttributes_FilterMetrics` test) from this pr - or if there's some way of 'ignore'ing tests in go, perhaps that would be appropriate? I feel like the changes to the `FilterMetricsByName` tests are appropriate, since they now correspond to the similarly named trace tests. Not exactly sure what cleanup would be suitable...

Reply:
Since it doesn't work, perhaps it's best just to delete that test set? Would appreciate some guidance on that, I don't know what's most likely to get this pr unblocked..

Reply:
Have opened https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/8395 since maybe that'll be a less painful approach

Reply:
> Have opened #8395 since maybe that'll be a less painful approach

Thanks @hughsimpson, I think we should fix the underlying issue rather than build on the broken piece of code